### PR TITLE
feat(txs): <- allow unrecognized tx types

### PR DESCRIPTION
**Pull request by gskapka**
Since the algorand network introduces new tx types surprisingly frequently, and the pTokens core doesn't user them, this PR allows them to become "unrecognized" without breaking anything.

### CODE REVIEW

File: src/algorand_transactions/transaction_type.rs [hunk 1]
```diff
@@ -51,12 +55,13 @@ impl FromStr for AlgorandTransactionType {
...
         match s {
             "pay" => Ok(Self::Pay),
+            "stpf" => Ok(Self::StateProof),
             "afrz" => Ok(Self::AssetFreeze),
             "axfer" => Ok(Self::AssetTransfer),
             "appl" => Ok(Self::ApplicationCall),
             "keyreg" => Ok(Self::KeyRegistration),
             "acfg" => Ok(Self::AssetConfiguration),
-            _ => Err(format!("Unrecognized Algorand tx type: '{s}'!").into()),
+            _ => Ok(Self::Unrecognized),
```
Comment (oliviera9) [inline src/algorand_transactions/transaction_type.rs:64]: That's the bypass!

File: src/algorand_blocks/block.rs [hunk 2]
```diff
@@ -276,4 +277,11 @@ mod test {
                 }
             })
     }
+
+    #[test]
+    fn should_parse_block_with_state_proof_tx() {
+        let json_string = get_sample_block_with_state_proof_tx_json_string();
+        let result = AlgorandBlock::from_str(&json_string);
+        assert!(result.is_ok());
```
Comment (oliviera9) [inline src/algorand_blocks/block.rs:285]: 👍

File: src/algorand_blocks/block_header.rs [hunk 3]
```diff
@@ -434,4 +442,13 @@ mod tests {
         let expected_result = "TOK6N7YCFFO27ALBWPJY6EMODHNX3IXM7ZKWTC7UPDE6XLEIFMQQ".to_string();
         assert_eq!(result, expected_result);
     }
+
+    #[test]
+    fn should_calculate_block_header_hash_3() {
+        let header = get_sample_block_header_n(10);
```
Comment (oliviera9) [inline src/algorand_blocks/block_header.rs:448]: Is this a block with the new `transactions_root_sha256` field?

**Review by oliviera9 — APPROVED** [review]
LGTM

File: src/algorand_transactions/transaction_type.rs (hunk 1 above)
Comment (gskapka) [reply src/algorand_transactions/transaction_type.rs:64]: Discussed in pvt & resolved.

**Pull request merged**